frontend/pages/doctor/appointments-overview.tsx

```diff
@@ -0,0 +1,132 @@
+            userId,
+        },
+    };
+}
+export default function AppointmentsOverview({
+    appointmentList,
+    patientList,
+    userId,
+}: {
+    appointmentList: Appointment[];
+    patientList: PatientBasicInformation[];
+    userId: number;
+}) {
+    const { onOpen, isOpen, onClose } = useDisclosure();
+
+    function sortAppointments(section: string): Appointment[] {
+        const upcomingAppointments: Appointment[] = [];
+        const pastAppointments: Appointment[] = [];
+        let appointmentDate: Moment = moment();
+
+        appointmentList.map((appointment: Appointment) => {
+            appointmentDate = moment(
+                `${appointment.date} ${appointment.time.substring(0, appointment.time.indexOf(" "))}`,
+                "YYYY-MM-DD HH:mm"
+            );
+            if (appointmentDate.isBefore()) {
+                pastAppointments.push(appointment);
+            } else if (appointmentDate.isAfter()) {
+                upcomingAppointments.push(appointment);
+            }
+        });
+        if (section === SECTION.UPCOMING) return upcomingAppointments;
+        else if (section === SECTION.PAST) return pastAppointments;
+        return appointmentList;
+    }
+    return (
+        <Box m={10}>
+            <Heading>Appointments</Heading>
+            <Flex flexDirection={{ base: "column", sm: "column", md: "column", lg: "row" }}>
+                {/* Calendar */}
+                <Box display={{ base: "none", md: "none", lg: "none", xl: "block", sm: "none" }} flex="1">
+                    <AppointmentCalendar appointmentList={appointmentList} />
+                </Box>
+                {/* Legend */}
+                <Box flex="1" h={650} minW={450}>
+                    <Box fontSize="lg" flex="1" h="10vh" mt={4}>
+                        <Text>
+                            <MinusIcon w={6} h={6} color="green.200" mr={3} /> Confirmed
+                        </Text>
+                        <Text flex="1">
+                            <MinusIcon w={6} h={6} color="orange.200" mr={3} /> Pending
+                        </Text>
+                        <Text flex="1">
+                            <MinusIcon w={6} h={6} color="red.200" mr={3} /> Denied
+                        </Text>
+                    </Box>
+                    <Box h={{ sm: "800", lg: "600" }}>
+                        {/* Upcoming appointments */}
+                        <Box>
+                            <Heading size="lg" my={2} display="inline-block">
+                                Upcoming
+                            </Heading>
+                            <Button variant="ghost" onClick={onOpen} float="right">
+                                <BsCalendarPlus style={{ width: "25px", height: "25px" }} />
+                            </Button>
+                            <Box h={{ sm: "400", md: "400", lg: "250" }} overflowX="hidden">
+                                <Scrollbars autoHide>
```

Comment:
Check for how this package works, it makes the appointments disappear in mobile view.

![image](https://user-images.githubusercontent.com/57551054/161144006-1ce27f71-91c0-4690-a76e-723f09729305.png)

Upcoming appointments here have the scrollbar component removed , but not the past appointments to see the difference

Reply:
I purposely made the calendar disappear on mobile cause the package used (for the calendar) is not that flexible with sizing lol

Reply:
Fixed